File: CORREDOR/Tipos Constructores/JS/Novela.js
const Libro = require("./Libro");

// Constructor con parámetros
class Novela extends Libro {
    constructor(titulo, autor, paginas, genero) {
        super(titulo, autor); // llama al constructor de la "clase abstracta"
        this.paginas = paginas;
        this.genero = genero;
    }

    abrir() {
        console.log("Abriendo la novela...");
    }

    informacion() {
        console.log(`Título: ${this.titulo}`);
        console.log(`Autor: ${this.autor}`);
        console.log(`Páginas: ${this.paginas}`);
        console.log(`Género: ${this.genero}`);
    }
}

const novela = new Novela("Cien años de soledad","Gabriel García Márquez",417,"Realismo mágico");
novela.abrir();
novela.informacion();

module.exports = Novela;

File: CORREDOR/Tipos Constructores/Java/Novela.java
package Java;

// Novela extiende Libro e implementa AccionesLibro.
public class Novela extends Libro implements AccionesLibro{
    int paginas;
    String genero;

    // Constructor con parámetros
    public Novela(String titulo, String autor, int paginas, String genero) {
        super(titulo, autor); // llama al constructor con parámetros de libro
        this.paginas = paginas;
        this.genero = genero;
    }

    @Override
    public void abrir() {
        System.out.println("Abriendo la novela...");
    }

    @Override
    void informacion() {
        System.out.println("Título: " + titulo);
        System.out.println("Autor: " + autor);
        System.out.println("Páginas: " + paginas);
        System.out.println("Género: " + genero);
    }
}